core/domain/exp_services.py

```diff
@@ -760,9 +761,12 @@ def delete_explorations(committer_id, exploration_ids, force_deletion=False):
     # Delete the explorations from search.
     search_services.delete_explorations_from_search_index(exploration_ids)
 
-    # Delete the exploration summaries, regardless of whether or not
-    # force_deletion is True.
+    # Delete the exploration summaries, recommendations and opportunities
+    # regardless of whether or not force_deletion is True.
     delete_exploration_summaries(exploration_ids)
+    recommendations_services.delete_explorations_from_recommendations(
+        exploration_ids)
+    opportunity_services.delete_exploration_opportunities(exploration_ids)
```

Comment:
Should we mark all the suggestions related to the exploration as rejected?

Reply:
Yep, I will do that in a different PR, I have this tracked in my wipeout list of task. Just to be sure, you mean `GeneralSuggestionModel`?

Reply:
Yes, the general suggestion mdoels!